src/Strategy/Python/Errors.hs

```diff
@@ -0,0 +1,44 @@
+module Strategy.Python.Errors (
+  MissingPoetryLockFile (..),
+  PipenvCmdFailed (..),
+
+  -- * docs
+  commitPoetryLockToVCS,
+) where
+
+import Data.String.Conversion (toText)
+import Data.Text (Text)
+import Diag.Diagnostic (ToDiagnostic, renderDiagnostic)
+import Effect.Exec (Command)
+import Path (Abs, File, Path)
+import Prettyprinter (Pretty (pretty), indent, viaShow, vsep)
+
+commitPoetryLockToVCS :: Text
+commitPoetryLockToVCS = "https://python-poetry.org/docs/basic-usage/#commit-your-poetrylock-file-to-version-control"
+
+newtype PipenvCmdFailed = PipenvCmdFailed Command
+instance ToDiagnostic PipenvCmdFailed where
+  renderDiagnostic (PipenvCmdFailed cmd) =
+    vsep
+      [ pretty $ "We could not perform pipenv graph analysis using, command:" <> toText cmd
+      , ""
+      , indent 2 $
+          vsep
+            [ "Ensure pipenv executable is in your PATH."
+            ]
+      ]
+
+newtype MissingPoetryLockFile = MissingPoetryLockFile (Path Abs File)
+instance ToDiagnostic MissingPoetryLockFile where
```

Comment:
I would be doing refactor of this type later, once I have seen all types of error cases into a some ActionableDiagCtx. But for now, I would like to get this in.